tests/ops/mapper/test_image_diffusion_mapper.py

```diff
@@ -0,0 +1,124 @@
+                    move_to_path = os.path.join(move_to_dir, os.path.basename(image_path))
+                    shutil.move(image_path, move_to_path)
+
+    def test_for_strength(self):
+        ds_list = [{
+            'text': f'{SpecialTokens.image}a photo of a cat',
+            'caption': f'a women with an umbrella',
+            'images': [self.cat_path]
+        }]
+        aug_num = 3
+        dataset = NestedDataset.from_list(ds_list)
+        op = ImageDiffusionMapper(
+            hf_diffusion=self.hf_diffusion,
+            strength=1.0,
+            aug_num=aug_num,
+            keep_original_sample=True,
+            caption_key='caption'
+        )
+        self._run_mapper(dataset, op, os.path.join(self.output_dir, 'test_for_strength'), total_num=(aug_num+1)*len(ds_list))
+
+
+    def test_for_given_caption_list(self):
+
+        ds_list = [{
+            'text': f'{SpecialTokens.image}, {SpecialTokens.image}',
+            'captions': ['A photo of a cat', 'a women with an umbrella'],
+            'images': [self.cat_path, self.img3_path]
+        }]
+
+        aug_num = 2
+        dataset = NestedDataset.from_list(ds_list)
+        op = ImageDiffusionMapper(
+            hf_diffusion=self.hf_diffusion,
+            aug_num=aug_num,
+            keep_original_sample=False,
+            caption_key='captions'
+        )
+        self._run_mapper(dataset, op, os.path.join(self.output_dir, 'test_for_given_caption_list'), total_num=aug_num*len(ds_list))
+
+    def test_for_given_caption_string(self):
+
+        ds_list = [{
+            'text': f'{SpecialTokens.image}a photo of a cat',
+            'images': [self.cat_path]
+        }, {
+            'text': f'{SpecialTokens.image}a photo, a women with an umbrella',
+            'images': [self.img3_path]
+        }]
+
+        aug_num = 1
+        dataset = NestedDataset.from_list(ds_list)
+        op = ImageDiffusionMapper(
+            hf_diffusion=self.hf_diffusion,
+            aug_num=aug_num,
+            keep_original_sample=False,
+            caption_key='text'
+        )
+        self._run_mapper(dataset, op, os.path.join(self.output_dir, 'test_for_given_caption_string'), total_num=aug_num*len(ds_list))
+
+    def test_for_no_given_caption(self):
+
+        ds_list = [{
+            'text': f'{SpecialTokens.image}',
+            'images': [self.cat_path]
+        }, {
+            'text': f'{SpecialTokens.image}',
+            'images': [self.img3_path]
+        }]
+
+        aug_num = 2
+        dataset = NestedDataset.from_list(ds_list)
+        op = ImageDiffusionMapper(
+            hf_diffusion=self.hf_diffusion,
+            aug_num=aug_num,
+            keep_original_sample=False,
+            hf_blip2=self.hf_blip2
+        )
+        self._run_mapper(dataset, op, os.path.join(self.output_dir, 'test_for_no_given_caption'), total_num=aug_num*len(ds_list))
+
+if __name__ == '__main__':
+    unittest.main()
```

Comment:
Recommended to add multiprocessing testing.

Reply:
Please check [prepare diffusion model and test](https://github.com/alibaba/data-juicer/pull/200/commits/3d5125dfb7aafad5f7853d52a49f24eab5d2fdea)